pkg/utils/builders.go

```diff
@@ -54,33 +87,25 @@ func DeployJob(perExperiment ExperimentDetails, engineDetails EngineDetails, env
 	return nil
 }
 
-// BuildPodTemplateSpec will build the PodTemplateSpec for further usage
-func BuildPodTemplateSpec(perExperiment ExperimentDetails, engineDetails EngineDetails, envVar []corev1.EnvVar, volumeMounts []corev1.VolumeMount, volumeBuilders []*volume.Builder) *podtemplatespec.Builder {
-
+// BuildPodTemplateSpec return a PodTempplateSpec
+func BuildPodTemplateSpec(perExperiment ExperimentDetails, engineDetails EngineDetails, envVar []corev1.EnvVar) *podtemplatespec.Builder {
 	podtemplate := podtemplatespec.NewBuilder().
 		WithName(perExperiment.JobName).
 		WithNamespace(engineDetails.AppNamespace).
 		WithLabels(perExperiment.ExpLabels).
 		WithServiceAccountName(engineDetails.SvcAccount).
-		WithVolumeBuilders(volumeBuilders).
-		WithRestartPolicy(corev1.RestartPolicyOnFailure).
-		WithContainerBuilders(
-			container.NewBuilder().
-				WithName(perExperiment.JobName).
-				WithImage(perExperiment.ExpImage).
-				WithCommandNew([]string{"/bin/bash"}).
-				WithArgumentsNew(perExperiment.ExpArgs).
-				WithImagePullPolicy("Always").
-				WithVolumeMountsNew(volumeMounts).
-				WithEnvsNew(envVar),
-		)
+		WithRestartPolicy(corev1.RestartPolicyOnFailure)
 	return podtemplate
 }
 
+// BuildJobSpec returns a JobSpec
 func BuildJobSpec(pod *podtemplatespec.Builder) *jobspec.Builder {
 	jobSpecObj := jobspec.NewBuilder().
 		WithPodTemplateSpecBuilder(pod)
-
+	_, err := jobSpecObj.Build()
```

Comment:
Why building `jobspec` here is required when its not being used??

Reply:
After your change, creating jobSpecObj is needed to use WithJobSpecBuilder.

Reply:
But you are returning `jobSpecObj` i.e a builder and not its object. Then why to build it anyway?

Reply:
Building is just to check for the errors, before deploying it.

Reply:
You can anyways get the errors if any while building the job at line 117

Reply:
It is always useful for debugging purpose, because kube-helper returns easier to decode errors.

Reply:
`Job's` build method will also throw the same error.

Reply:
It will just send this error failed to build job: nil templatespecbuilder , and not that the error is in podTemplateSpec

Reply:
I suppose there is a requirement to have custom errors printed out along with validation to check the feasibility of building the object (esp. when comprised of multiple sub-objects) -- before we actually use it to create/deploy in runtime. Some kind of a builderValidator function that can be not only applied in the jobSpec case but any other as well -- this way it can be reused across code.